react/introduction/fundamentals/react-what-is-state.md

```diff
@@ -117,7 +114,10 @@ Where should **initial state** be set in a component class?
 ## Footnotes
 
 [1:Immutable State]
-This has to do with how data flows in React. The general idea behind this concept is that data goes down while actions go up, effectively creating a circular system. This will be explained in more detail and the end of the workout.
+This has to do with how data flows in React. The general idea behind this concept is that data goes down while actions go up, effectively creating a circular system. This will be explained in more detail at the end of the workout.
 
 [2:Changing state before mounting]
-If you try to change the `state` before the component was mounted, React will display the value of the initial `state`.
\ No newline at end of file
+If you try to change the `state` before the component was mounted, React will display the value of the initial `state`.
+
+[3:Memory Leaks]
+This is mostly due to how JavaScript works, rather than React. In JS, closures can live past the lifetime of a function, and since an interval callback needs to be kept alive by the browser to get executed, it can accidentally keep other references via its closure. Thus, memory isn't cleanep up, no matter if the React component is there or not. Here's [an article](https://reliablejavascript.com/2017/11/05/how-timer-intervals-can-leak-memory/) that provides a more in-depth reading.
```

Comment:
```suggestion
This is mostly due to how JavaScript works, rather than React. In JS, closures can live past the lifetime of a function, and since an interval callback needs to be kept alive by the browser to get executed, it can accidentally keep other references via its closure. Thus, memory isn't cleaned up, no matter if the React component is there or not. Here's [an article](https://reliablejavascript.com/2017/11/05/how-timer-intervals-can-leak-memory/) that provides a more in-depth reading.
```